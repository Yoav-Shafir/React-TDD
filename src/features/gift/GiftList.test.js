import React from "react";
import { shallow, mount } from "enzyme";
import renderer from "react-test-renderer";

import GiftListContainer from "./GiftList.container";
import GiftListComponent from "./GiftList.component";

describe("<GiftListContainer />", () => {
  let container;
  beforeEach(() => {
    container = shallow(<GiftListContainer />);
  });
  it("initializes the `state` with an empty `gifts` list", () => {
    expect(container.state().gifts).toEqual([]);
  });
  describe("`addGift()` and `removeGift()`", () => {
    beforeEach(() => container.instance().addGift());
    it("adds a gift to `state`", () => {
      expect(container.state().gifts.length).toEqual(1);
      expect(container.state().gifts).toEqual([{ id: 1 }]);
    });
    it("removes a gift from `state`", () => {
      container.instance().removeGift(1);
      expect(container.state().gifts.length).toEqual(0);
      expect(container.state().gifts).toEqual([]);
    });
  });
});

describe("<GiftListComponent />", () => {
  const mockAddGift = jest.fn();
  const mockRemoveGift = jest.fn();
  const props = {
    gifts: [{ id: 1 }, { id: 2 }],
    addGift: mockAddGift,
    removeGift: mockRemoveGift
  };
  const component = mount(<GiftListComponent {...props} />);
  it("renders correctly", () => {
    const component = renderer
      .create(<GiftListComponent {...props} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  describe("when clicking the `Add Gift` button", () => {
    it("executes `addGift()`", () => {
      component.find("button.btn-add").simulate("click");
      expect(mockAddGift).toBeCalled();
    });
  });
});
