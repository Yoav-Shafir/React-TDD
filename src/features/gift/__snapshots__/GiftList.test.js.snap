// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GiftListComponent /> renders correctly 1`] = `
Array [
  <h2>
    Gift Giver
  </h2>,
  <div
    className="gift-list"
  >
    <div
      className="gift"
    >
      <form
        className=""
      >
        <div
          className="form-group"
        >
          <label
            className="form-label"
          >
            Person
          </label>
          <input
            className="input-person form-control"
            onChange={[Function]}
          />
        </div>
        <div
          className="form-group"
        >
          <label
            className="form-label"
          >
            Present
          </label>
          <input
            className="input-present form-control"
            onChange={[Function]}
          />
        </div>
      </form>
      <button
        className="btn-remove btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Remove Gift
      </button>
    </div>
    <div
      className="gift"
    >
      <form
        className=""
      >
        <div
          className="form-group"
        >
          <label
            className="form-label"
          >
            Person
          </label>
          <input
            className="input-person form-control"
            onChange={[Function]}
          />
        </div>
        <div
          className="form-group"
        >
          <label
            className="form-label"
          >
            Present
          </label>
          <input
            className="input-present form-control"
            onChange={[Function]}
          />
        </div>
      </form>
      <button
        className="btn-remove btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Remove Gift
      </button>
    </div>
  </div>,
  <button
    className="btn-add btn btn-primary"
    disabled={false}
    onClick={[MockFunction]}
    type="button"
  >
    Add Gift
  </button>,
]
`;
